library/src/context.cxx

```diff
@@ -0,0 +1,143 @@
+#include "context.h"
+
+#include <vtksys/DynamicLoader.hxx>
+
+namespace f3d
+{
+//----------------------------------------------------------------------------
+context::function context::glx()
+{
+#ifdef __linux__
+  std::string libName = vtksys::DynamicLoader::LibPrefix();
+  libName += "GLX";
+  libName += vtksys::DynamicLoader::LibExtension();
+
+  vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(libName);
+
+  if (!handle)
+  {
+    throw loading_exception("Cannot find GLX library");
+  }
+
+  using symbol = fptr (*)(const char*);
+
+  symbol address =
+    reinterpret_cast<symbol>(vtksys::DynamicLoader::GetSymbolAddress(handle, "glXGetProcAddress"));
+
+  if (!address)
+  {
+    throw symbol_exception("Cannot find glXGetProcAddress symbol");
+  }
+
+  return address;
+#else
+  throw loading_exception("Cannot use a GLX context on this platform");
+#endif
+}
+
+//----------------------------------------------------------------------------
+context::function context::egl()
+{
+#ifndef __APPLE__
```

Comment:
why not on apple btw ?

Reply:
Idk that might be possible but not natively (maybe XQuart is needed?)